### ISSUE TITLE
RecylableMemoryStreamManager.GetStream should return RMS instead of MS

### ISSUE THREAD

**Issue by benmwatson** [microsoft/Microsoft.IO.RecyclableMemoryStream#165]
Verify binary compatibility is maintained.

**Comment by grbell-ms** [comment]
I was mistaken, binary compatibility is not maintained.
https://docs.microsoft.com/en-us/dotnet/core/compatibility/#properties-fields-parameters-and-return-values

**Comment by benmwatson** [comment]
Yeah, I've run into this before. Thanks for double-checking. It doesn't automatically make this change a no-go, but the bar is certainly higher. Might be good for v3.

**Comment by stefan2410** [comment]
@benmwatson   because of that issue,  can't get the  Write(ReadOnlySpan<byte> source) as I had mentioned in #162 ?
